### ISSUE TITLE
R connection using library RPostgres fails with error "received invalid response to GSSAPI negotiation: R"

### ISSUE THREAD

**Issue by mskyttner** [ArcadeData/arcadedb#227]
## Expected behavior

Attempt to connect using R and `RPostgres` expected to work, since RJDBC is able to read data from a table in arcadedb.

## Actual behavior

Error `"received invalid response to GSSAPI negotiation: R"`

## Steps to reproduce

```r

arcadedb_cmd <- paste(
  "docker run -d -p '2480:2480' -p '2424:2424' -p '5432:5432'",
  "--env arcadedb.server.rootPassword=playwithdata",
  "--env 'arcadedb.server.defaultDatabases=Imported[root]{import:https://github.com/ArcadeData/arcadedb-datasets/raw/main/orientdb/OpenBeer.gz}'", 
  "--env arcadedb.server.plugins='Postgres:com.arcadedb.postgres.PostgresProtocolPlugin'", 
  "arcadedata/arcadedb:latest"
)

cmd <- gsub("\n", "", arcadedb_cmd)

# launch the background service with arcadedb using docker
system(arcadedb_cmd)

# check that it is online using the API
arcadedb_is_up <- function() {
  
  res <- httr::content(
    httr::GET("http://localhost:2480/api/v1/databases", 
    httr::authenticate("root", "playwithdata"))
  )
  
  res$result[[1]] == "Imported"
}

stopifnot(arcadedb_is_up())

# This "works", using RJDBC and the pg jdbc driver

download.file(
  "https://jdbc.postgresql.org/download/postgresql-42.3.1.jar",
  "~/repos/arcadedb/postgresql-42.3.1.jar"
)

library(RJDBC)

drv <- JDBC(
  driverClass = "org.postgresql.Driver", 
  classPath = "~/repos/arcadedb/postgresql-42.3.1.jar", 
  identifier.quote = "`"
)

conn <- dbConnect(drv, 
  paste0("jdbc:postgresql://localhost/Imported",
    "?user=root&password=playwithdata&ssl=false&gssencmode=disable"))

tibble::as_tibble(dbGetQuery(conn, "select * from Brewery limit 10;"))

dbDisconnect(conn)

# Now, attempt to use RPostgres which uses libpg and no Java

library(RPostgres)
library(DBI)

#Sys.setenv(PGGSSENCMODE="disable")
Sys.unsetenv("PGGSSENCMODE")

con <- dbConnect(
  RPostgres::Postgres(),
#  gssencmode = 'disable',
  dbname = 'Imported',
  host = 'localhost',
  port = 5432, 
  user = 'root',
  password = 'playwithdata'
)

# Error: connection to server at "localhost" (127.0.0.1), port 5432 failed: received invalid response to GSSAPI negotiation: R
# Can this be worked around by disabling GSSENCMODE?
# see https://stackoverflow.com/questions/67396081/how-to-configure-a-odbc-connection-to-teiid-on-rhel8-gssapi-negotiation-fails
# and https://blog.hagander.net/connecting-to-azure-postgresql-with-libpq-12-in-a-kerberos-environment-245/

```

**Comment by lvca** [comment]
Is it possible is the same issue as for https://githubmemory.com/repo/r-dbi/RPostgres/issues/310?

Please disable SSL with the connection: it's not supported yet but it's low priority assuming you're connecting client and server in a local (trusted) network.

**Comment by mskyttner** [comment]
I logged an issue with the RPostgres package and made an attempt to disable SSL but the connection still "hangs". 

https://github.com/r-dbi/RPostgres/issues/363

For me, the OpenBeer database running in under docker uses 2.2 GB of RAM (from `docker stats`). Are there any docs regarding memory limits for arcadedb, say if I wanted to run it with only 1 or a 0.5 GB of RAM, are there any settings/parameters that could be tweaked to allow this?

**Comment by lvca** [comment]
You're right, our docs lack this information. I'm gonna improve it today. For now I report here my comment on R-DB issue:
------------
You can set in ArcadeDB's docker `-Darcadedb.profile=low-ram` to run ArcadeDB with <32M of RAM and see how it goes. In general, the RAM allocated for the JVM should be <=80% of the container RAM.

The default Dockerfile sets `2GB` of RAM for ArcadeDB (`-Xms2G -Xmx2G`), so you should allocate around 2.3G to the Docker container. 

To run ArcadeDB with 1G docker container, you could start ArcadeDB by using 800M for RAM:

```
docker .. -e ARCADEDB_OPTS_MEMORY="-Xms800M -Xmx800M" ...
```
------------

**Comment by lvca** [comment]
Added this paragraph hoping helps a little more, any suggestions to improve it are welcome: https://docs.arcadedb.com/#DockerTuning

**Comment by lvca** [comment]
@mskyttner fixed in the latest "main" branch and `22.1.1-SNAPSHOT` (now the `latest`) on Docker Hub. Please let me know if works for you. I tested with the following script and works.

```R
.libPaths( c( "/usr/local/lib/R/lib" , .libPaths() ) )

r = getOption("repos")
r["CRAN"] = "http://cran.us.r-project.org"
options(repos = r)

##install.packages(c("RPostgres"), lib="/usr/local/lib/R/lib")
##install.packages(c("DBI"), lib="/usr/local/lib/R/lib")

library(DBI)

#Sys.setenv(PGGSSENCMODE="disable")
#Sys.unsetenv("PGGSSENCMODE")
#Sys.unsetenv("PGSSLMODE")
#Sys.setenv(PGSSLMODE="disable")

con <- dbConnect(
  RPostgres::Postgres(),
  gssencmode = 'disable',
  sslmode = "disable",
  dbname = 'Beer',
  host = 'localhost',
  port = 5432,
  user = 'root',
  password = 'playwithdata'
)

print("EXECUTING QUERY")

res <- dbSendQuery(con, "select * from Brewery limit 10")

print("FETCHING RESULTS")

dbFetch(res)

print("CLEARING RESULTS")
dbClearResult(res)
dbDisconnect(con)
```

Now I have to find a way to include the R test suite in our CI pipeline... @arcade-player any help?
